code/modules/makeup/nailpolish.dm

```diff
@@ -0,0 +1,158 @@
+/obj/item/organ/external/var/datum/nail_polish/nail_polish
+
+/obj/item/weapon/nailpolish
+	name = "nail polish"
+	desc = "Nail polish, to paint your nails with. Or someone else's!"
+	icon = 'icons/obj/nailpolish_vr.dmi'
+	icon_state = "nailpolish"
+	var/colour = "#FF0000"
+	var/icon/top_underlay
+	var/icon/top_underlay_open
+	var/icon/color_underlay
+	var/icon/color_underlay_open
+	var/open = FALSE
+	drop_sound = 'sound/items/drop/glass.ogg'
+	pickup_sound = 'sound/items/pickup/glass.ogg'
+
+/obj/item/weapon/nailpolish/Initialize()
+	. = ..()
+	regenerate_icons()
+
+/obj/item/weapon/nailpolish/proc/regenerate_icons()
+	top_underlay = icon(icon, "top")
+	top_underlay_open = icon(icon, "top-open")
+	color_underlay = icon(icon, "color")
+	color_underlay.Blend(colour, ICON_MULTIPLY)
+	color_underlay_open = icon(icon, "color-open")
+	color_underlay_open.Blend(colour, ICON_MULTIPLY)
+	update_icon()
```

Comment:
Icon manipulation is realllllyyyyy slow and CPU intensive on the server. I'd suggest making these mutable appearances (use the `mutable_appearance()` helper) and setting the `color` var. The byond default use of the `color` var is multiply, so you should achieve the same result.